detect_secrets/core/scan.py

```diff
@@ -323,6 +330,11 @@ def _process_line_based_plugins(
             line_number=line_number,
         )
 
+        raw_code_snippet = get_code_snippet(
+            lines=read_raw_lines(filename),
```

Comment:
@marynaKK Do we want the entire file here? It could be very heavy... Maybe choose a subset of the lines?

Reply:
I thought about it - I wanted to create a similar object as context. The most prune-to-errors way to do it is to use the same logic, the function of `get_code_snippet`. Otherwise, It will require playing with indexes and could lead to a mismatch between the 2 objects.
The following code is from bc-detect-secrets read file logic. It is similar to what I did here, so I hope that if it was working till now, maybe it can be acceptable:

```
    @property
    def raw_lines(self) -> List[str]:
        if self._raw_lines:
            return self._raw_lines

        with self.open_file() as f:
            self._raw_lines = [line.rstrip() for line in f.readlines()]

        return self._raw_lines
```

Reply:
Sounds good